### DISCUSSION TITLE
About creatStore for Streaming

### DISCUSSION THREAD

**Discussion by 19Qingfeng** [pmndrs/zustand#2053] [Q&A]
## problem

I have encountered some writing problems and did not find any relevant instructions here after consulting the documentation. So I had to explore it myself, but I don't actually think it's a bug. But I think the missing part of the document is equally important.

When I use SSR with Streaming doc, I cannot directly use React Create Api. Instead, I can only create a store in the Suspend through createStore in conjunction with React.createContext, which is passed through the context after the streaming is completed.

I'm not sure if there's a better solution, but currently this is the solution I can think of. If I create a store before streaming and update the store after streaming is completed. At this point, it will cause the entire component to be re rendered, and then the streaming component will be re executed.

error case: 

<img width="1260" alt="image" src="https://github.com/pmndrs/zustand/assets/31119562/80227683-1d50-497e-bc17-b0723e3a1c94">

I had to modify the code to make it look like this

<img width="1355" alt="image" src="https://github.com/pmndrs/zustand/assets/31119562/808de36f-dd59-4d72-ad65-e3edbb6b6d5c">

But in this way, every time I use it, I need to make non null judgments, and I feel like I have returned to the way reactContext is used

I would like to inquire if there is a more suitable and elegant way to handle the coordination issue of Streaming. If only this way is available, perhaps I can propose some of my own ideas for secondary packaging and updating the document

I am very willing to contribute my PR, if there is currently no unified solution for such issues in Zustad.

You can reproduce it here [repo](https://github.com/19Qingfeng/zustand-remix)
1. npm install
2. npm dev
3. open http://localhost:3000/error    you can see console multiple render.
4. open http://localhost:3000             success but it's redundant.

**Comment by dbritto-dev** [comment]
@19Qingfeng please check this out -> https://docs.pmnd.rs/zustand/guides/initialize-state-with-props

**Comment by 19Qingfeng** [comment]
> @19Qingfeng please check this out -> https://docs.pmnd.rs/zustand/guides/initialize-state-with-props

Excuse me, can I only use context? In this case, I still call the create method after the data is returned after initialization. Not createStore and contenxt.

**Comment by dbritto-dev** [comment]
@19Qingfeng if you want to initialize your state `context` + `createStore` is the right way, try not to use `create` in that case.
